sklearn/ensemble/iforest.py

```diff
@@ -206,7 +219,9 @@ def predict(self, X):
 
         depths += _average_path_length(n_samples_leaf)
 
-        if not isinstance(self.max_samples, (numbers.Integral, np.integer)):
```

Comment:
Here something is wrong (and was wrong before your changes): we should use the `max_samples` passed to the `base_estimator` (and in this case we don't need those 3 checks).
Use directly `max_samples = self.base_estimator.max_samples` ?